packages/react-router/src/route.ts

```diff
@@ -722,50 +732,42 @@ export class Route<
     const isRoot = !options?.path && !options?.id
 
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    this.parentRoute = this.options?.getParentRoute?.()
+    const parentRoute = options?.getParentRoute?.()
+
+    let path: undefined | string
+    let customId: string | undefined
+    let id: string
+    let fullPath: string
 
     if (isRoot) {
-      this.path = rootRouteId as TPath
+      path = '/'
+      customId = options?.id || rootRouteId
+      id = rootRouteId
+      fullPath = '/'
     } else {
       invariant(
-        this.parentRoute,
+        parentRoute,
         `Child Route instances must pass a 'getParentRoute: () => ParentRoute' option that returns a Route instance.`,
       )
-    }
-
-    let path: undefined | string = isRoot ? rootRouteId : options.path
 
-    // If the path is anything other than an index path, trim it up
-    if (path && path !== '/') {
-      path = trimPathLeft(path)
+      path = options.path ? trimPathLeft(options.path) : undefined
+      customId = options.id || path
+      id = joinPaths([
+        '/',
+        parentRoute.id === rootRouteId ? '' : parentRoute.id,
+        customId,
+      ])
+      fullPath = joinPaths([parentRoute.fullPath, path])
     }
 
-    const customId = options?.id || path
-
-    // Strip the parentId prefix from the first level of children
-    let id = isRoot
-      ? rootRouteId
-      : joinPaths([
-          this.parentRoute.id === rootRouteId ? '' : this.parentRoute.id,
-          customId,
-        ])
-
-    if (path === rootRouteId) {
-      path = '/'
+    return {
+      parentRoute: parentRoute as TParentRoute,
+      path: path as TPath,
+      id: id as TId,
+      // customId: customId as TCustomId,
```

Comment:
I assume the original `customId` stuff was commented out (instead of just removed) for a reason, so I kept it in as a comment.